src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Services/IUserTimeZoneService.cs

```diff
@@ -0,0 +1,25 @@
+using OrchardCore.Modules;
+
+namespace OrchardCore.Users.TimeZone.Services;
+
+/// <summary>
+/// Contract for user time zone service.
+/// </summary>
+public interface IUserTimeZoneService
+{
+    /// <summary>
+    /// Gets the time zone of the currently logged-in user.
+    /// </summary>
+    public ValueTask<ITimeZone> GetUserTimeZoneAsync();
```

Comment:
I would rename this method to GetTimeZoneAsync(IUser user). I would also pass IUser and parameter so it's not current user. You can add an extension for the IUserTimeZoneService that would take HttpContext and return current user's timezone. This way the interface is more useful and does not contain implementation details.